File: examples/key/sign/sign.js
const {
  AuthenticityClient,
  EcdsaSigner,
  KeyClient,
  RecordClient,
  KeyType,
} = require('@bloock/sdk');

(async () => {
  let keyClient = new KeyClient();
  let authenticityClient = new AuthenticityClient();
  let recordClient = new RecordClient();

  let key = await keyClient.newLocalKey(KeyType.EcP256k);

  let signedRecord = await recordClient
    .fromString('Hello world')
    .withSigner(new EcdsaSigner(key))
    .build();

  console.log('Record was signed successfully');

  // we can add another signature with a different key
  key = await keyClient.newLocalKey(KeyType.EcP256k);

  console.log('Adding another signature');
  signedRecord = await recordClient
    .fromRecord(signedRecord)
    .withSigner(new EcdsaSigner(key))
    .build();

  console.log('Record was signed successfully');

  console.log(`Hash: ${await signedRecord.getHash()}`);

  let signatures = await authenticityClient.getSignatures(signedRecord);
  signatures.forEach((signature, i) => {
    console.log(`Signature ${i + 1}: ${JSON.stringify(signature, null, 4)}`);
  });
})();

File: examples/key/sign/sign.go
package main

import (
	"encoding/json"
	"fmt"
	"log"

	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity/authenticity"
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
)

func main() {
	keyClient := client.NewKeyClient()
	recordClient := client.NewRecordClient()
	authenticityClient := client.NewAuthenticityClient()

	localKey, err := keyClient.NewLocalKey(key.EcP256k)
	if err != nil {
		log.Println(err)
	}

	signedRecord, err := recordClient.FromString("Hello world").
		WithSigner(authenticity.NewSignerWithLocalKey(localKey, nil)).
		Build()
	if err != nil {
		log.Println(err)
	}
	fmt.Println("Record was signed successfully")

	// we can add another signature with a different key
	localKey, err = keyClient.NewLocalKey(key.EcP256k)
	if err != nil {
		log.Println(err)
	}

	fmt.Println("Adding another signature")
	signedRecord, err = recordClient.FromRecord(signedRecord).
		WithSigner(authenticity.NewSignerWithLocalKey(localKey, nil)).
		Build()
	if err != nil {
		log.Println(err)
	}
	fmt.Println("Record was signed successfully")

	hash, err := signedRecord.GetHash()
	if err != nil {
		log.Println(err)
	}
	fmt.Printf("Hash: %+v", hash)

	signatures, err := authenticityClient.GetSignatures(signedRecord)
	if err != nil {
		log.Println(err)
	}

	for i, signature := range signatures {
		prettySignature, _ := json.MarshalIndent(signature, "", "    ")
		fmt.Printf("Signature %d: %s\n", i+1, string(prettySignature))
	}
}
